tools/clang/test/CodeGenSPIRV/meshshading.ext.no-explicit-exts-spv1.4.mesh.hlsl

```diff
@@ -0,0 +1,33 @@
+// RUN: %dxc -T ms_6_5 -E main -fspv-target-env=vulkan1.1spirv1.4 %s -spirv | FileCheck %s
+// CHECK:  OpCapability MeshShadingEXT
+// CHECK:  OpExtension "SPV_EXT_mesh_shader"
+// CHECK:  OpEntryPoint MeshEXT %main "main" %gl_GlobalInvocationID %gl_Position [[primind:%[0-9]+]]
```

Comment:
```suggestion
// CHECK:  OpEntryPoint MeshEXT %main "main" %gl_Position [[primind:%[0-9]+]]
```

Reply:
That's weird because gl_GlobalInvocationID isn't optimized out for me, so the test fails locally for me.
Is there a way to do an optional regex in filecheck? like:
// CHECK:  OpEntryPoint MeshEXT %main "main" (%gl_GlobalInvocationID)? %gl_Position [[primind:%[0-9]+]]
Also I don't know if 2 whitespace vs. 1 will matter above ^, I might need to squish it:
(%gl_GlobalInvocationID )?%gl_Position
I can't find a reference for the syntax on effcee

Reply:
I figured out how to do a regex here. I made the %gl_GlobalInvocationID optional in the entry point. The tests should pass now

Reply:
Why is it optional? The tests should be deterministic on all platforms. Do we understand why there is a difference?

Reply:
No Idea.
After removing -fcgl, %gl_GlobalInvocationID is there for me locally, but optimized out on the github runner

Reply:
Let me try rebasing or cloning a fresh repo locally

Reply:
rebasing to top of main didn't seem to help

Reply:
I'll try taking a closer look to figure out why there is a difference. Did you update the submodules after rebasing? I think that https://github.com/KhronosGroup/SPIRV-Tools/pull/5551 could change the result of this test.

Reply:
Ah we were looking at the same thing 🙃
It's definitely the submodules.

```
git submodule deinit -f .
git submodule update --init
```

Reply:
I tested it out myself, and everything said here is consistent with the submodules not be updated. That is the only difference I can think of.

Reply:
Thanks, submodules makes sense.
I think I've screwed up my local repo's badly. I tried to rebase and squash my commits onto main but they end up in between random commits from main.
If there's an easier way for you to rebase and squash from the UI when you accept the PR that'd be great, using the message of the first commit.

I pushed a final commit that removes the regex from the tests.

Reply:
Thanks @rj12122! I've rebased your branch to get rid of the merge commit and force pushed to update the PR. Hopefully the auto-merge will work now.